src/backend/distributed/commands/utility_hook.c

```diff
@@ -252,8 +252,12 @@ citus_ProcessUtility(PlannedStmt *pstmt,
 		 * and if the command is a node-wide object management command that we
 		 * support from non-main databases.
 		 */
+		bool shouldSkipPrevUtilityHook = false;
 
-		bool shouldSkipPrevUtilityHook = RunPreprocessNonMainDBCommand(parsetree);
+		if (EnableDDLPropagation)
```

Comment:
Maybe we can inline EnableDDLPropagation checks into RunPre/PostprocessNonMainDBCommand functions as we do for IsMainDb checks within those functions?

Reply:
Fixed as you suggested